src/dmd/dsymbolsem.d

```diff
@@ -2018,6 +2023,69 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
          */
         addEnumMembers(ed, sc, sc.getScopesym());
 
+        if (sc.flags & SCOPE.Cfile)
+        {
+            /* C11 6.7.2.2
+             */
+            ed.memtype = Type.tint32; // C11 6.7.2.2-4 implementation defined
+            int nextValue = 0;        // C11 6.7.2.2-3 first member value defaults to 0
+
+            void emSemantic(EnumMember em, ref int nextValue)
```

Comment:
```suggestion
            static void emSemantic(EnumMember em, ref int nextValue)
```
You're passing `em` and `nextValue` explicitly, rather than going via a closure/frame.  Might as well make the nested function static to enforce that the context pointer really isn't needed.

Reply:
The function uses `sc` too.